internal/observability/event/eventer.go

```diff
@@ -59,8 +61,18 @@ type broker interface {
 	RegisterPipeline(def eventlogger.Pipeline) error
 }
 
+// queuedEvent stores an event and the context that was associated with it when
+// gating
+type queuedEvent struct {
+	ctx   context.Context
+	event interface{}
+}
+
 // Eventer provides a method to send events to pipelines of sinks
 type Eventer struct {
+	gated                atomic.Bool
```

Comment:
perhaps a comment on these new fields stating that gating is limited to error/sys events?

Reply:
Done.